src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -0,0 +1,206 @@
+                            case SplittingStrategy.PerRun:                           
+                            default:
+                                throw new ArgumentOutOfRangeException($"GroupingStrategy: {splittingStrategy}");
+                        }
+
+                        visitor.VisitRun(sarifLog.Runs[runIndex]);
+                        logsToProcess = visitor.SplitSarifLogs;
+                    }
+
+                    for (int splitFileIndex = 0; splitFileIndex < logsToProcess.Count; splitFileIndex++)
+                    {
+                        SarifLog splitLog = logsToProcess[splitFileIndex];
+                        FileWorkItemsForSarifLog(sarifLog);
+                    }
+                }
+            }
+        }
+
+        private void FileWorkItemsForSarifLog(SarifLog sarifLog)
+        {
+            // First we make a copy of the pipeline root context. This allows us to produce
+            // new context files, potentially with outputs, that are specific to each log.
+            
+            SarifWorkItemModel workItemModel = sarifLog.CreateWorkItemModel(this.FilingContext);
+
+            // Now we initialize this work item model from the SARIF log file driving the
+            // current operations. As a first step, this code should extract all 
+            // source-controlled files from the log file. In an imminent implementation
+            // this information will be used to derive area path and owner information.
+            
+            workItemModel.Context.InitializeFromLog(sarifLog);
+
+            try
+            {
+                // Populate the work item with the target organization/repository information.
+                // In ADO, certain fields (such as the area path) will defaut to the 
+                // project name and so this information is used in at least that context. 
+                workItemModel.RepositoryOrProject = this.FilingClient.ProjectOrRepository;
+                workItemModel.OwnerOrAccount = this.FilingClient.AccountOrOrganization;
+
+                foreach (SarifWorkItemModelTransformer transformer in workItemModel.Context.Transformers)
+                {
+                    transformer.Transform(workItemModel);
+                }
+
+                this.FilingClient.FileWorkItems(new[] { workItemModel }).Wait();
+
+                // TODO: We need to process updated work item models to persist filing
+                //       details back to the input SARIF file, if that was specified.
+                //       This code should either return or persist the updated models
+                //       via a property, so that the file work items command can do
+                //       this work.
+                //
+                //       https://github.com/microsoft/sarif-sdk/issues/1774
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine(ex.ToString());
```

Comment:
>.ToString() [](start = 42, length = 11)

This can be removed, internally WriteLine(Object value) already calls value.ToString()

Reply:
That's true and I'll delete it. I'm in the habit of explicitly calling ToString() on things to avoid calling into an object accepting method. The reason is that the underlying code may utilize an unexpected CultureInfo/IFormatProvider (such as the UI specific vs. invariant culture). This problem doesn't have any relevance to the exception object, though.

---
In reply to: [381453913](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381453913) [](ancestors = 381453913)

Reply:
"The reason is that the underlying code may utilize an unexpected CultureInfo/IFormatProvider (such as the UI specific vs. invariant culture)" 

Cool. I didn't think about this scenario. Thanks for pointing that out.

---
In reply to: [383399024](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r383399024) [](ancestors = 383399024,381453913)